### PR TITLE
Nebula Agent for br

### PR DESCRIPTION
**Pull request by pengweisong**
Nebula Agent is an daemon service in each machine of [nebula](https://github.com/vesoft-inc/nebula) cluster. It helps to keep track of nebula metad/storaged/graphd, start/stop them, call local rpc of them.

### CODE REVIEW

File: internal/clients/daemon.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	d := &ServiceDaemon{
+		s: s,
+	}
+	return d, nil
+}
+
+func (d *ServiceDaemon) Start() error {
+	cmdStr := fmt.Sprintf("cd %s && scripts/nebula.service start %s", d.s.dir, d.s.name)
+	log.WithField("cmd", cmdStr).Debug("Try to start service")
+	cmd := exec.Command("bash", "-c", cmdStr)
```
[the remark this answers is not in the document]
Comment (pengweisong) [reply internal/clients/daemon.go:78]: It's risk indeed. The code will only check the scripts return code. And when return code is 0, it does not mean the service has must been started successfully.

[the remark this answers is not in the document]
Comment (pengweisong) [reply internal/clients/daemon.go:78]: I'll add a TODO

File: internal/clients/meta.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+}
+
+func (m *NebulaMeta) heartbeat() error {
+	req := &meta.AgentHBReq{
+		Host:       m.config.AgentAddr,
+		GitInfoSha: []byte(m.config.GitInfoSHA),
+	}
+
+	// retry only when leader change
+	for {
```
[the remark this answers is not in the document]
Comment (pengweisong) [reply internal/clients/meta.go:123]: It only retry when the leader change and will stop when encounter other error

**Review by critical27 — DISMISSED** [review]

File: internal/clients/meta.go [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+	if resp.GetCode() != nebula.ErrorCode_SUCCEEDED {
+		return nil, fmt.Errorf("get meta dir info failed: %v", resp.GetCode())
+	}
+
+	return resp.GetDir(), nil
+}
+
+// GetServiceInfo return the given sevice's info which agent keep monitoring
+func (m *NebulaMeta) GetServiceInfo(addr *nebula.HostAddr) *meta.ServiceInfo {
```
[the remark this answers is not in the document]
Comment (pengweisong) [reply internal/clients/meta.go:195]: I'll remove it

File: README.md [hunk 4]
```diff
@@ -1 +1,44 @@
...
+
+## File Management
+
+```proto
+// UploadFile upload file from agent machine to external storage
+rpc UploadFile(UploadFileRequest) returns (UploadFileResponse);
+// DownloadFile download file from external storage to agent machine
+rpc DownloadFile(DownloadFileRequest) returns (DownloadFileResponse);
+// MoveDir rename dir in agent machine
+rpc MoveDir(MoveDirRequest) returns (MoveDirResponse);
```
[the remark this answers is not in the document]
Comment (pengweisong) [reply README.md:15]: Yes~

**Review by Nivras — APPROVED** [review]

**Review by critical27 — APPROVED** [review]

**Pull request merged**